### ISSUE TITLE
LDAP support

### ISSUE THREAD

**Issue by yagarea** [Arksine/moonraker#388]
### Is your feature request related to a problem? Please describe

It would be nice to be able to authenticate to moonraker with LDAP.

### Describe the solution you'd like

Implementation of LDAP support into moonraker

**Comment by mjparme** [comment]
why?

**Comment by Preisschild** [comment]
I'd also be interested in this (or SAML or OAuth)

Useful if you already have a central authentication service and dont want to manage users over multiple locations.

Also using solutions like this allows logging who logged in when.

**Comment by yagarea** [comment]
> why?

Our university has it as hard condition for network managed software to have LDAP authtentication. Otherwise we are forbidden to install it on school machines.

**Comment by lucaschoeneberg** [comment]
Hi, I added a Pull Request for this Issue: https://github.com/Arksine/moonraker/pull/436

**Comment by Arksine** [comment]
FYI, I had merged #436, however I had to quickly revert as the dependency failed to build on some systems, resulting in a failure to load the `authorization` module.   I pushed a modified version with a pure python dependency.  The ldap configuration and functionality is in its own module, so even if it fails to load the `authorization` module will still load correctly.  This also makes it easier to expand ldap authentication options, for example it would be relatively easy to add support for ssl certs or sasl authentication.
